thoth/storages/graph/postgres.py

```diff
@@ -7050,25 +7050,59 @@ def get_python_package_version_all(self, analysis_document_id: str) -> List[Dict
     def get_python_package_version_import_packages_all(
         self, import_name: str, distinct: bool = False
     ) -> List[Dict[str, str]]:
-        """Retrieve Python package name for the given import package name."""
+        """Retrieve Python package name for the given import package name.
+
+        Examples:
+        >>> from thoth.storages import GraphDatabase
+        >>> graph = GraphDatabase()
+        >>> graph.get_python_package_version_import_packages_all("faust.*")
+        [
+           {'import': 'faust.web.apps',
+            'index_url': 'https://pypi.org/simple',
+            'package_name': 'faust',
+            'package_version': '1.9.0'},
+           {'import': 'faust.web.cache',
+            'index_url': 'https://pypi.org/simple',
+            'package_name': 'faust',
+            'package_version': '1.9.0'},
+           {'import': 'faust.web.cache.backends',
+            'index_url': 'https://pypi.org/simple',
+            'package_name': 'faust',
+            'package_version': '1.9.0'},
+           {'import': 'faust.web.drivers',
+            'index_url': 'https://pypi.org/simple',
+            'package_name': 'faust',
+            'package_version': '1.9.0'}
+            ...
+        ]
+        """
         with self._session_scope() as session:
+            query = session.query(PythonPackageVersion).join(FoundImportPackage).join(ImportPackage)
+
+            if import_name.endswith("*"):
+                query = query.filter(ImportPackage.import_package_name.like(f"{import_name[:-1]}%"))
+            else:
+                query = query.filter(ImportPackage.import_package_name == import_name)
+
             query = (
-                session.query(PythonPackageVersion)
-                .join(FoundImportPackage)
-                .join(ImportPackage)
-                .filter(ImportPackage.import_package_name == import_name)
-                .filter(PythonPackageVersion.id == FoundImportPackage.python_package_version_id)
+                query.filter(PythonPackageVersion.id == FoundImportPackage.python_package_version_id)
                 .join(PythonPackageIndex)
                 .filter(PythonPackageVersion.python_package_index_id == PythonPackageIndex.id)
+                .distinct()
```

Comment:
do we introduce it by default?

Reply:
I'll remove it as the parameter is already propagated on user api. BTW not sure about the applicability of `distinct` here - the database should keep each record unique anyway.

Reply:
> 

the distinct is because of the solver, we receive twice the same package otherwise: https://github.com/thoth-station/storages/blob/0b0e38359369efdb3e6547eba3debf48b1c25d39/thoth/storages/graph/models.py#L64